general/front-end-security-guidelines.md

```diff
@@ -0,0 +1,37 @@
+### HTTPS
+Make sure all of your resources are being loaded through a secured HTTPS connection and all client-server communication is happening through an HTTPS protocol. Also, keep an eye out for an unsecured connections initiated by a third-party dependencies.
+
+### User input validation and escaping to prevent XSS
+Data that is being provided by the user should always be strictly validated and escaped before sent to the server. Make sure there are no possible scenarios for XSS through user generated data. To prevent this kind of vulnerabilities measures should be taken both at the server side and on the client.
+
+### Using untrusted data in the DOM (DomXSS)
+Any time you need to insert some data directly to the DOM, make sure to do it in the right way. 
+If you’re using data from the server (especially 3rd-party), it should not be received as an HTML ready to be inserted to the DOM straight away. The preferred way is to send the data in JSON format, parse and validate it on the client side and then use the data as required.
+If you have to put some data provided by the user in the DOM, make sure it’s properly validated and sanitised (doesn’t contain any HTML, styles or scripts in it) before using it. Some modern frameworks will do that for you but not all of them, so keep that in mind. 
+Every time you have to use dangerouslySetInnerHTML in your frontend framework, dynamically place user input into a.href, style or other HTML attributes that support “javascript:” option, think twice and try to avoid doing that.
+
+### Avoid eval()
+Not only is code using eval() is highly prone to error, but it’s also a dangerous security vulnerability. There’s very little chance you really need to use eval() to make your code work as desired, so, please, avoid it.
+
+### Broken authentication
+There’s a number of things could go wrong with your authentication flow. Let’s go through the most common scenarios.
+#### Putting Session ID in the URL (ex: http://example.com/&sessionid=3FAN8DKFNVMDPOSLAFK7R)
+Doing so you're exposing the session identifier, furthermore, if the user shares the link with somebody the’ll use his session. Luckily this is a pretty old technique and almost nobody does it nowadays, but you have to be aware of it just in case.
```

Comment:
```suggestion
This exposes the session identifier, furthermore, if the user shares the link with somebody they’ll use his session. Luckily this is a pretty old technique and almost nobody does it nowadays, but you have to be aware of it just in case.
```